src/hdfs.rs

```diff
@@ -171,6 +190,39 @@ impl HdfsFs {
         self.raw
     }
 
+    pub fn connect_with_config(
+        nn: &str,
+        config: &HashMap<String, String>,
+    ) -> Result<Arc<Self>, HdfsErr> {
```

Comment:
The main entry for getting an `Arc<HdfsFs>` is `get_hdfs_by_full_path`. It caches already connected file system handles and will reuse them when getting another `HdfsFs` using uri with cached host. `HdfsFs` handles were never closed, since the Rust binding does not call `hdfsDisconnect` at all.

The caller of `connect_with_config` has to deal with caching and reusing already connected `HdfsFs`, otherwise we'll leave lots of connected Hadoop FileSystem objects in the JVM.